pkg/apis/pipeline/v1/pipelinerun_defaults.go

```diff
@@ -37,7 +37,11 @@ func (pr *PipelineRun) SetDefaults(ctx context.Context) {
 func (prs *PipelineRunSpec) SetDefaults(ctx context.Context) {
 	cfg := config.FromContextOrDefaults(ctx)
 
-	if prs.Timeouts != nil && prs.Timeouts.Pipeline == nil {
+	if prs.Timeouts == nil {
+		prs.Timeouts = &TimeoutFields{}
+	}
+
+	if prs.Timeouts.Pipeline == nil {
```

Comment:
I think this would be more readable as

```
if prs.Timeouts == nil || prs.Timeouts.Pipeline == nil {
     prs.Timeouts.Pipeline = &metav1.Duration{Duration: time.Duration(cfg.Defaults.DefaultTimeoutMinutes)  * time.Minute}
}

Reply:
Thanks for the suggestion. In this case seems thiere could be potential null ptr 🤔

Reply:
Can you elaborate? In the case where `prs.Timeouts` is nil, I think this should be fine because only the first block of the "or" will be evaluated.

Reply:
https://go.dev/ref/spec#Logical_operators: "The right operand is evaluated conditionally"